enterprise/server/workflow/service/service.go

```diff
@@ -627,10 +630,18 @@ func (ws *workflowService) createActionForWorkflow(ctx context.Context, wf *tabl
 	}
 	conf := ws.env.GetConfigurator()
 	containerImage := ""
+	isolationType := ""
 	os := strings.ToLower(workflowAction.OS)
 	// Use the CI runner image if the OS supports containerized actions.
 	if os == "" || os == platform.LinuxOperatingSystemName {
 		containerImage = ws.workflowsImage()
+		if *enableFirecracker {
+			isolationType = string(platform.FirecrackerContainerType)
+			// When using Firecracker, write all outputs to the scratch disk, which
+			// has more space than the workspace disk and doesn't need to be extracted
+			// to the executor between action runs.
+			envVars = append(envVars, &repb.Command_EnvironmentVariable{Name: "WORKDIR_OVERRIDE", Value: "/root/workspace"})
+		}
```

Comment:
would this be better to happen on the executor side, or do we not modify the action there?

Reply:
We don't modify the action on the executor, so it's probably cleaner to keep this logic here to avoid more special cases for workflow actions

Reply:
mostly just thinking it's a tad annoying to have to know ahead of time, in the app, that we're going to be using firecracker in the executor and override something like the workdir which we have no context about. Feels like if this were doable on the executor, right before firecracker execution, that might be better. But I may be misunderstanding!

Reply:
(discussed offline, i can live with this)